src/libcrun/signals.c

```diff
@@ -595,14 +595,20 @@ str2sig (const char *name)
     {
       long int value;
 
+      if (!isdigit(name[0]))
+        goto error;
+
       errno = 0;
       value = strtol (name, NULL, 10);
-      if (errno == 0)
-        return value;
+      if (errno != 0)
+        goto error;
```

Comment:
and use `return -1;` here

Reply:
Should I set errno in here before return?